core/src/main/java/com/amplifyframework/auth/options/AuthSignInOptions.java

```diff
@@ -16,83 +16,21 @@
 package com.amplifyframework.auth.options;
 
 import androidx.annotation.NonNull;
-import androidx.core.util.ObjectsCompat;
-
-import com.amplifyframework.util.Immutable;
-
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Objects;
-
-public final class AuthSignInOptions {
-    private final Map<String, String> validationData;
-
-    /**
-     * Advanced options for signing in.
-     * @param validationData A map of custom key/values to be sent as part of the sign in process
-     */
-    private AuthSignInOptions(Map<String, String> validationData) {
-        this.validationData = validationData;
-    }
-
-    /**
-     * Get a map of custom key/values to be sent as part of the sign in process.
-     * @return a map of custom key/values to be sent as part of the sign in process
-     */
-    @NonNull
-    public Map<String, String> getValidationData() {
-        return validationData;
-    }
-
-    @Override
-    public int hashCode() {
-        return ObjectsCompat.hash(
-                getValidationData()
-        );
-    }
-
-    @Override
-    public boolean equals(Object obj) {
-        if (this == obj) {
-            return true;
-        } else if (obj == null || getClass() != obj.getClass()) {
-            return false;
-        } else {
-            AuthSignInOptions authSignInOptions = (AuthSignInOptions) obj;
-            return ObjectsCompat.equals(getValidationData(), authSignInOptions.getValidationData());
-        }
-    }
-
-    @Override
-    public String toString() {
-        return "AuthSignInOptions{" +
-                "validationData=" + validationData +
-                '}';
-    }
-
-    @NonNull
-    public static Builder builder() {
-        return new Builder();
-    }
-
-    public static final class Builder {
-        private Map<String, String> validationData;
-
-        public Builder() {
-            this.validationData = new HashMap<>();
-        }
-
-        @NonNull
-        public Builder validationData(@NonNull Map<String, String> validationData) {
-            Objects.requireNonNull(validationData);
-            this.validationData.clear();
-            this.validationData.putAll(validationData);
-            return this;
-        }
 
+// Note: This is currently empty but exists here to support common sign in options in the future.
+public abstract class AuthSignInOptions {
```

Comment:
An abstract class that has no implementation and no policy is really just like an interface. Can this, and its inner-Builder be interfaces, instead?

Interfaces are always better than abstract classes, since Java doesn't support multiple inheritence of classes, but you can implement as many interfaces as you want.

The only exception is when you _have_ to use a class, because you want to enforce policy. And in that category:
1. Use a final class with composition. Don't need to worry about multiple inheritence - much less any inheritence, if the class is closed.
2. Use an abstract class that acts like an interface, but constrains what the extensible methods are able to achieve, through hard-coded policy.

Reply:
I debated this in my head and chose to go this route because this is a direct mirror of AuthSignUpOptions and very much expect to have common options go in here just like in it's counterpart AuthSignUpOptions so want to have it implemented in such a way that there is minimal code impact to the rest of the library when we do that.